In [3]:
import pandas  as pd
df = pd.read_csv('brief_assurance.csv')
df.head(10)

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520
5   31  female  25.740         0     no  southeast   3756.62160
6   46  female  33.440         1     no  southeast   8240.58960
7   37  female  27.740         3     no  northwest   7281.50560
8   37    male  29.830         2     no  northeast   6406.41070
9   60  female  25.840         0     no  northwest  28923.13692

In [4]:
print(df.isnull().sum())


age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64


In [5]:
df.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

selon les descriptions stastique, nous n'avons pas constacte les valeur aberrants ou manquantes. pour max bmi de 53.13 correspondre a un homme de 180cm et 170kg. reste une valeur non aberrants. même conculusion pour nombre d'enfants et age .
donc notre dataframe est prêt à emplois

In [10]:
df_charge_decroissant = df.sort_values("charges", ascending=False)
df_charge_decroissant.head(20)


age     sex     bmi  children smoker     region      charges
543    54  female  47.410         0    yes  southeast  63770.42801
1300   45    male  30.360         0    yes  southeast  62592.87309
1230   52    male  34.485         3    yes  northwest  60021.39897
577    31  female  38.095         1    yes  northeast  58571.07448
819    33  female  35.530         0    yes  northwest  55135.40209
1146   60    male  32.800         0    yes  southwest  52590.82939
34     28    male  36.400         1    yes  southwest  51194.55914
1241   64    male  36.960         2    yes  southeast  49577.66240
1062   59    male  41.140         1    yes  southeast  48970.24760
488    44  female  38.060         0    yes  southeast  48885.13561
175    63  female  37.700         0    yes  southwest  48824.45000
530    57    male  42.130         1    yes  southeast  48675.51770
901    60    male  40.920         0    yes  southeast  48673.55880
281    54    male  40.565         3    yes  northeast  48549.17835
330    61  female  36.385         1    yes  northeast  48517.56315
39     60    male  39.900         0    yes  southwest  48173.36100
328    64  female  33.800         1    yes  southwest  47928.03000
288    59  female  36.765         1    yes  northeast  47896.79135
55     58    male  36.955         2    yes  northwest  47496.49445
951    51    male  42.900         2    yes  southeast  47462.89400

In [11]:
df_charge_croissant = df.sort_values("charges")
df_charge_croissant.head(20)

age   sex    bmi  children smoker     region    charges
940    18  male  23.21         0     no  southeast  1121.8739
808    18  male  30.14         0     no  southeast  1131.5066
1244   18  male  33.33         0     no  southeast  1135.9407
663    18  male  33.66         0     no  southeast  1136.3994
22     18  male  34.10         0     no  southeast  1137.0110
194    18  male  34.43         0     no  southeast  1137.4697
866    18  male  37.29         0     no  southeast  1141.4451
781    18  male  41.14         0     no  southeast  1146.7966
442    18  male  43.01         0     no  southeast  1149.3959
1317   18  male  53.13         0     no  southeast  1163.4627
1251   19  male  19.80         0     no  southwest  1241.5650
681    19  male  20.30         0     no  southwest  1242.2600
584    19  male  20.70         0     no  southwest  1242.8160
791    19  male  27.60         0     no  southwest  1252.4070
487    19  male  28.70         0     no  southwest  1253.9360
513    19  male  30.40         0     no  southwest  1256.2990
136    19  male  34.10         0     no  southwest  1261.4420
385    19  male  34.40         0     no  southwest  1261.8590
723    19  male  35.40         0     no  southwest  1263.2490
374    20  male  33.33         0     no  southeast  1391.5287